lib/HttpClient.ts

```diff
@@ -186,48 +198,62 @@ export class HttpClient implements ifm.IHttpClient {
+        let response: HttpClientResponse;
+        while (numTries > 0) {
+            let response: HttpClientResponse = await this.requestRaw(info, data);
+
+            // Check if it's an authentication challenge
+            if (response && response.message && response.message.statusCode === HttpCodes.Unauthorized) {
+                let authenticationHandler: ifm.IRequestHandler;
+
+                for (let i = 0; i < this.handlers.length; i++) {
+                    if (this.handlers[i].canHandleAuthentication(response)) {
+                        authenticationHandler = this.handlers[i];
+                        break;
+                    }
                 }
-            }
 
-            if (authenticationHandler) {
-                return authenticationHandler.handleAuthentication(this, info, data);
-            }  
-            else {
-                // We have received an unauthorized response but have no handlers to handle it.
-                // Let the response return to the caller.
-                return response;
+                if (authenticationHandler) {
+                    return authenticationHandler.handleAuthentication(this, info, data);
+                }  
+                else {
+                    // We have received an unauthorized response but have no handlers to handle it.
+                    // Let the response return to the caller.
+                    return response;
+                }
             }
-        }
 
-        let redirectsRemaining: number = this._maxRedirects;
-        while (HttpRedirectCodes.indexOf(response.message.statusCode) != -1
-               && this._allowRedirects
-               && redirectsRemaining > 0) {
+            let redirectsRemaining: number = this._maxRedirects;
+            while (HttpRedirectCodes.indexOf(response.message.statusCode) != -1
+                && this._allowRedirects
+                && redirectsRemaining > 0) {
+
+                const redirectUrl: any = response.message.headers["location"];
+                if (!redirectUrl) {
+                    // if there's no location to redirect to, we won't
+                    break;
+                }
+
+                // we need to finish reading the response before reassigning response
+                // which will leak the open socket.
+                await response.readBody();
 
-            const redirectUrl: any = response.message.headers["location"];
-            if (!redirectUrl) {
-                // if there's no location to redirect to, we won't
-                break;
+                // let's make the request with the new redirectUrl
+                info = this._prepareRequest(verb, redirectUrl, headers);
+                response = await this.requestRaw(info, data);
+                redirectsRemaining--;
             }
 
-            // we need to finish reading the response before reassigning response
-            // which will leak the open socket.
-            await response.readBody();
```

Comment:
I think we need to keep this.

Reply:
Yeah, its still there, the diff is just displaying differently. Its on line 243 and occurs in the same place relative to the original code.